### PR TITLE
Promtool: add web config validation

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

**Comment by roidelapluie** [comment]
This is ready for review.  cc @brian-brazil @simonpasquier

File: cmd/promtool/main.go [hunk 1]
```diff
@@ -234,6 +244,25 @@ func CheckConfig(files ...string) int {
...
+	failed := false
+
+	for _, f := range files {
+		err := https.Validate(f)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, f, "FAILED:", err)
+			failed = true
+			continue
+		}
+		fmt.Fprintln(os.Stderr, f, "SUCCESS")
```
Comment (bwplotka) [inline cmd/promtool/main.go:258]:
```suggestion
		if err := https.Validate(f); err != nil {
			fmt.Fprintln(os.Stderr, f, "FAILED:", err)
			failed = true
			continue
		}
		fmt.Fprintln(os.Stderr, f, "SUCCESS")
```

File: cmd/promtool/main.go [hunk 2]
```diff
@@ -59,6 +60,12 @@ func main() {
 		"The config files to check.",
 	).Required().ExistingFiles()
 
+	checkWebConfigCmd := checkCmd.Command("web-config", "Check if the web config files are valid or not.")
```
Comment (bwplotka) [inline cmd/promtool/main.go:63]: Why web config is not part of Prometheus config? 🤔

**Review by bwplotka — APPROVED** [review]
LGTM, just one question, maybe already discussed somewhere.

Thanks!

File: cmd/promtool/main.go (hunk 2 above)
Comment (bwplotka) [reply cmd/promtool/main.go:63]: ok got my answer: https://github.com/prometheus/prometheus/pull/8316/files#diff-114e09a49c240c764f484c12b5f54019aafcfbe1a71f820d1c6b192ede1d5d18R13

**Review by bwplotka — APPROVED** [review]

**Pull request merged**